### PR TITLE
Remove PodSecurityPolicy admission plugin

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup
/kind deprecation

#### What this PR does / why we need it:

Removes the PodSecurityPolicy admission plugin

#### Which issue(s) this PR fixes:
xref https://github.com/kubernetes/enhancements/issues/5

#### Does this PR introduce a user-facing change?
```release-note
The beta `PodSecurityPolicy` admission plugin, deprecated since 1.21, is removed. Follow the instructions at https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/ to migrate to the built-in PodSecurity admission plugin (or to another third-party policy webhook) prior to upgrading to v1.25.
```

/sig auth security
/cc @deads2k @enj @ritazh @mikedanese

### CODE REVIEW

File: pkg/apis/policy/validation/validation.go [hunk 1]
```diff
@@ -33,8 +33,15 @@ import (
 	core "k8s.io/kubernetes/pkg/apis/core"
 	apivalidation "k8s.io/kubernetes/pkg/apis/core/validation"
 	"k8s.io/kubernetes/pkg/apis/policy"
-	"k8s.io/kubernetes/pkg/security/podsecuritypolicy/seccomp"
-	psputil "k8s.io/kubernetes/pkg/security/podsecuritypolicy/util"
+)
+
+const (
```
Comment (liggitt) [inline pkg/apis/policy/validation/validation.go:38]: these were hoisted from k8s.io/kubernetes/pkg/security/podsecuritypolicy/seccomp to allow deleting that package

File: pkg/apis/policy/validation/validation.go [hunk 2]
```diff
@@ -332,6 +339,44 @@ func validatePodSecurityPolicyVolumes(fldPath *field.Path, volumes []policy.FSTy
 	return allErrs
 }
 
+// getAllFSTypesAsSet returns all actual volume types, regardless
+// of feature gates. The special policy.All pseudo type is not included.
+func getAllFSTypesAsSet() sets.String {
```
Comment (liggitt) [inline pkg/apis/policy/validation/validation.go:344]: this was hoisted from k8s.io/kubernetes/pkg/security/podsecuritypolicy/util to allow deleting that package

File: pkg/kubelet/sysctl/safe_sysctls.go [hunk 3]
```diff
@@ -16,13 +16,17 @@ limitations under the License.
...
-import (
-	"k8s.io/apimachinery/pkg/util/validation/field"
-	api "k8s.io/kubernetes/pkg/apis/core"
-)
-
-// SysctlsStrategy defines the interface for all sysctl strategies.
-type SysctlsStrategy interface {
-	// Validate ensures that the specified values fall within the range of the strategy.
-	Validate(pod *api.Pod) field.ErrorList
+// SafeSysctlAllowlist returns the allowlist of safe sysctls and safe sysctl patterns (ending in *).
```
Comment (liggitt) [inline pkg/kubelet/sysctl/safe_sysctls.go:19]: this was hoisted from k8s.io/kubernetes/pkg/security/podsecuritypolicy/sysctl to allow deleting that package

**Comment by enj** [comment]
/triage accepted
/milestone v1.25
/lgtm
/hold
To allow others to review.

**Review by enj — APPROVED** [review]

**Comment by deads2k** [comment]
heh, yeah, test-cmd got me in the 1.25 removals.

/lgtm

File: pkg/apis/policy/validation/validation.go [hunk 4]
```diff
@@ -33,8 +33,15 @@ import (
...
-	psputil "k8s.io/kubernetes/pkg/security/podsecuritypolicy/util"
+)
+
+const (
+	// AllowAny is the wildcard used to allow any profile.
+	seccompAllowAny = "*"
+	// DefaultProfileAnnotationKey specifies the default seccomp profile.
+	seccompDefaultProfileAnnotationKey = "seccomp.security.alpha.kubernetes.io/defaultProfileName"
+	// AllowedProfilesAnnotationKey specifies the allowed seccomp profiles.
+	seccompAllowedProfilesAnnotationKey = "seccomp.security.alpha.kubernetes.io/allowedProfileNames"
```
Comment (sftim) [inline pkg/apis/policy/validation/validation.go:44]: These are being removed too, right? As part of https://github.com/kubernetes/kubernetes/pull/109819

Comment (liggitt) [reply pkg/apis/policy/validation/validation.go:44]: This API code is staying unchanged for 1.25 and will be removed in 1.26. The other PR is on my radar to review as well

**Pull request merged**